src/Symfony/Component/DependencyInjection/Dumper/XmlDumper.php

```diff
@@ -298,6 +298,10 @@ private function convertParameters(array $parameters, $type, \DOMElement $parent
                 $element->setAttribute('type', 'expression');
                 $text = $this->document->createTextNode(self::phpToXml((string) $value));
                 $element->appendChild($text);
+            } elseif (is_string($value) && preg_match('/[\x00-\x08\x0b-\x1f\x7f-\xff]/', $value)) {
```

Comment:
What about `!preg_match('//u', $value)` instead?

Reply:
I tried `/\p{C}/u` but it was not working as expected

Reply:
I retested it and tried to detect control characters instead except whitespaces